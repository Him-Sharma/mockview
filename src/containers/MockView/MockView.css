.MockView {
  color: white;
}
.StubContainer {
  display: flex;
  flex-direction: row;
  margin: 40px auto;
}
.Control * {
  background-color: white;
  font: inherit;
  border: 1px solid green;
  padding: 8px;
  margin: auto 10px;
}
.StubItemList {
  border: 2px solid #09d3ac;
  color: white;
  flex-basis: 30%;
  margin: 20px 20px;
}
