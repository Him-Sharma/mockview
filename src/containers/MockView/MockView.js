import React, { useEffect, useState } from "react";
import axios from "axios";
import Aux from "../../hoc/Aux";
import Control from "../../components/Control/Control";
import StubList from "../../components/StubList/StubList";

const MockView = () => {
  const [stubs, setStubs] = useState([]);
  const [mockServerURL, setMockServerURL] = useState("");

  useEffect(() => {
    if (mockServerURL) {
      axios.get(mockServerURL).then((response) => {
        console.log(response);
        const rawStubItems = response.data.stubs;
        const updatedStubs = rawStubItems.map((stubItem) => {
          return {
            ...stubItem,
            id: stubItem._links.self.href.split("/").pop(),
          };
        });
        console.log(updatedStubs);
        setStubs(updatedStubs);
      });
    }
  }, [mockServerURL]);
  const urlSubmitHandler = (mockServerURL) => {
    setMockServerURL(mockServerURL);
  };

  return (
    <Aux>
      <Control onFormSubmit={urlSubmitHandler} />
      <StubList stubs={stubs} />
    </Aux>
  );
};
export default MockView;
