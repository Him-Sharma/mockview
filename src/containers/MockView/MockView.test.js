import React from "react";
import { render, fireEvent } from "@testing-library/react";
import MockView from "./MockView";
import axios from "axios";

jest.mock("axios");

it("renders correctly", () => {
  const { baseElement } = render(<MockView />);
  expect(baseElement).toMatchSnapshot();
});

it("should dispatch request on click of fetch stubs button", () => {
  const testUrl = "http://test/5000";
  const response = {
    stubs: [
      {
        predicates: [
          {
            equals: {
              method: "GET",
              path: "/posts",
            },
          },
        ],
      },
    ],
  };
  axios.get.mockResolvedValue({
    data: response,
  });

  const { getByTestId } = render(<MockView />);

  //act
  fireEvent.change(getByTestId("serverUrlInput"), {
    target: { value: testUrl },
  });
  fireEvent.click(getByTestId("fetchStubButton"));

  //assert
  expect(axios.get).toHaveBeenCalledTimes(1);
  expect(axios.get).toHaveBeenCalledWith(testUrl);
});
