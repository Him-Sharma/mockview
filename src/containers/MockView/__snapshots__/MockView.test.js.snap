// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<body>
  <div>
    <div
      class="MockView"
    >
      <h1>
        Mockview
      </h1>
      <section
        class="Control"
      >
        <input
          data-testid="serverUrlInput"
          id="serverUrlInput"
          type="text"
        />
        <button
          data-testid="fetchStubButton"
        >
          Fetch Stubs
        </button>
      </section>
      <section
        class="StubContainer"
      >
        <h1>
          No Stub items to display
        </h1>
      </section>
    </div>
  </div>
</body>
`;
